runtime/src/validate_block/implementation.rs

```diff
@@ -215,15 +215,16 @@ impl<B: BlockT> Storage for WitnessStorage<B> {
 			Err(_) => panic!(),
 		};
 
-		let mut iter = trie.iter().expect("Creates trie iterator");
-		iter.seek(prefix).expect("Seek trie iterator");
+		let iterator = match TrieDBIterator::new_prefixed(&trie, prefix)
+		{
+			Ok(r) => r,
+			Err(_) => panic!(),
+		};
 
-		for x in iter {
+		for x in iterator {
 			let (key, _) = x.expect("Iterating trie iterator");
 
-			if !key.starts_with(prefix) {
-				break;
-			}
+			debug_assert!(!key.starts_with(prefix));
```

Comment:
This code will probably never compiled with debug asserts enabled.

Reply:
That's sad, I am not sure how it differs from assert, but I am just removing it.

Reply:
`debug_assert!` is only activated on debug builds or when you set `debug-asserts`. However, I assume that we would hit the bug before on the node.